plugins/samples/overwrite_header/plugin.cc

```diff
@@ -15,21 +15,32 @@
 // [START serviceextensions_plugin_overwrite_header]
 #include "proxy_wasm_intrinsics.h"
 
+// This sample replaces an HTTP header with the given key and value.
+// Unlike `addRequestHeader` which appends values to existing headers,
+// this plugin overwrites the entire value for the specified key.
 class MyHttpContext : public Context {
  public:
   explicit MyHttpContext(uint32_t id, RootContext* root) : Context(id, root) {}
 
   FilterHeadersStatus onRequestHeaders(uint32_t headers,
                                        bool end_of_stream) override {
     // Change the key and value according to your needs
-    replaceRequestHeader("RequestHeader", "changed");
+    auto header_key = "RequestHeader";
+    auto header = getRequestHeader(header_key);
+    if (header->size() > 0) {
+      replaceRequestHeader(header_key, "changed");
+    }
     return FilterHeadersStatus::Continue;
   }
 
   FilterHeadersStatus onResponseHeaders(uint32_t headers,
                                         bool end_of_stream) override {
     // Change the key and value according to your needs
-    replaceResponseHeader("ResponseHeader", "changed");
+    auto header_key = "ResponseHeader";
+    auto header = getResponseHeader(header_key);
+    if (header->size() > 0) {
+      replaceResponseHeader(header_key, "changed");
+    }
```

Comment:
Works fine, but this identical example doesn't really teach the reader anything new. This might be an opportunity to call replaceResponseHeader without the existence check, and show (in a comment and in the tests) how that behavior is different (adds the header if it's missing).

The fact that "replace" refers to the value and not the key is the most surprising behavior of replace*Header(), so I think we should call it out clearly in the sample.